src/librustc/diagnostics.rs

```diff
@@ -176,6 +176,38 @@ for the entire lifetime of a program. Creating a boxed value allocates memory on
 the heap at runtime, and therefore cannot be done at compile time.
 "##,
 
+E0011: r##"
+Using a user-defined operator on const/static variable is restricted to what
```

Comment:
This first sentence doesn't really make sense as you can't use user-defined operators on consts or statics at all. I think we should accentuate the point from the second sentence. Something like:

> Initializers for constants and statics are evaluated at compile time. User-defined operators rely on user-defined functions, which cannot be evaluated at compile time.